yb-voyager/src/namereg/namereg.go

```diff
@@ -50,42 +70,43 @@ type NameRegistry struct {
 	DefaultSourceReplicaDBSchemaName string
 	// Source replica has same table name list as original source.
 
-	// Private members are not saved in the JSON file.
-	filePath string
-	role     string
-	sconf    *srcdb.Source
-	sdb      srcdb.SourceDB
-	tconf    *tgtdb.TargetConf
-	tdb      tgtdb.TargetDB
+	params NameRegistryParams
+}
+
+func InitNameRegistry(
+	exportDir string, role string,
+	sourceDbType string, sourceDbSchema string, sourceDbName string, targetDbSchema string,
+	sdb SourceDbRegistry,
+	ybdb YBDBRegistry) error {
```

Comment:
How about taking `params` as argument? Let the call construct the params object? The whole point of introducing the `params` object was to tame the list of arguments to the `NameRegistry` constructor.